domain/modelconfig/service/service.go

```diff
@@ -125,6 +132,59 @@ func (s *Service) ModelConfig(ctx context.Context) (*config.Config, error) {
 	return config.New(config.NoDefaults, altConfig)
 }
 
+// deserializeMap converts a map[string]string to map[string]any and coerces
+// any provider-specific values that are found in the provider's schema.
+func (s *Service) deserializeMap(m map[string]string) (map[string]any, error) {
+	result := make(map[string]any, len(m))
+
+	// If we don't have a model config provider getter, just do basic string->any conversion
+	if s.modelConfigProviderGetterFunc == nil {
+		return transform.Map(m, func(k, v string) (string, any) { return k, v }), nil
+	}
+
+	// Get the cloud type from the config
+	cloudType, ok := m[config.TypeKey]
+	if !ok || cloudType == "" {
+		// No cloud type - just convert without coercion
+		return transform.Map(m, func(k, v string) (string, any) { return k, v }), nil
+	}
+
+	// Get the provider for this cloud type
+	provider, err := s.modelConfigProviderGetterFunc(cloudType)
+	if err != nil {
+		// Provider not found or doesn't support schema - graceful degradation
+		return transform.Map(m, func(k, v string) (string, any) { return k, v }), nil
+	}
+
+	if provider == nil {
+		// No provider available - just convert without coercion
+		return transform.Map(m, func(k, v string) (string, any) { return k, v }), nil
+	}
+
+	// Get the schema from the provider
+	fields := provider.ConfigSchema()
+	if fields == nil {
+		// No schema available - just convert without coercion
+		return transform.Map(m, func(k, v string) (string, any) { return k, v }), nil
+	}
```

Comment:
I think we should just go through the coercion.

Reply:
In controllerconfig we simply ignore the keys that cannot be coerced https://github.com/juju/juju/blob/cbb05654c7444dd6bee29e49aff16339f02c34f9/domain/controllerconfig/service/service.go#L187-L196 for example.